src/shims/posix/foreign_items.rs

```diff
@@ -14,12 +15,15 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
     fn emulate_foreign_item_by_name(
         &mut self,
         link_name: &str,
+        abi: Abi,
         args: &[OpTy<'tcx, Tag>],
         dest: PlaceTy<'tcx, Tag>,
         ret: mir::BasicBlock,
     ) -> InterpResult<'tcx, bool> {
         let this = self.eval_context_mut();
 
+        check_abi(abi, Abi::C)?;
```

Comment:
So, for foreign items you just enforce that all Posix symbols use the `C` ABI... can you do the same for `Dlsym`, instead of pushing things down? This reduces the risk of forgetting to add the check there. ;)

Reply:
> So, for foreign items you just enforce that all Posix symbols use the C ABI... can you do the same for Dlsym, instead of pushing things down? This reduces the risk of forgetting to add the check there. ;)

Definitely ;) Wasn't sure if there would be non-C calls in either of the Posix Dlsyms. All fixed